scripts/globals/mobs.lua

```diff
@@ -31,46 +31,111 @@ local function lotteryPrimed(phList)
     return false
 end
 
-xi.mob.updateNMSpawnPoint = function(mob, spawnPoints)
-    -- This function is used to replace UpdateNMSpawnPoints() inside the Zone.lua files and the NM despawn scripts
-    -- Once UpdateNMSpawnPoints() is no longer used, this note can be removed
-    -- Spawnpoints is a table of {x = , y = , z = }
+local getMobLuaPathObject = function(mob)
+    if not mob then
+        return nil
+    end
+
+    return xi.zones[mob:getZoneName()].mobs[mob:getName()]
+end
+
+-- This function is used to replace UpdateNMSpawnPoints() inside the Zone.lua files and the NM despawn scripts
+-- - mobParam can either be a mobid or a mob entity object
+-- it either accepts a table of spawn points to randomize, or looks to the mob's cached lua object for a spawnPoints entry
+xi.mob.updateNMSpawnPoint = function(mobParam, spawnPointsOverride)
+    local mob = nil
+    -- sometimes we call from Zone.lua files and only have the mob id
+    if type(mobParam) == 'number' then
+        mob = GetMobByID(mobParam)
+    else
+        mob = mobParam
+    end
+
+    if not mob then
+        print('[updateNMSpawnPoint] Invalid mob parameter:')
+        print(mobParam)
+
+        return
+    end
+
+    -- if no spawnPoints table was sent, extract from mob lua object
+    local spawnPoints = spawnPointsOverride
+    if spawnPoints == nil then
+        local mobObject = getMobLuaPathObject(mob)
+        if mobObject and mobObject.spawnPoints then
+            spawnPoints = mobObject.spawnPoints
+        end
+    end
+
+    -- Special check for NMs with the same name but multiple IDs
+    if spawnPoints and spawnPoints[mob:getID()] then
+        spawnPoints = spawnPoints[mob:getID()]
+    end
+
     if spawnPoints ~= nil and #spawnPoints > 0 then
         local chosenSpawn    = utils.randomEntry(spawnPoints)
         local randomRotation = math.random(0, 255) -- rotation does not matter
 
         -- Updates the mob's spawn point
         mob:setSpawn(chosenSpawn.x, chosenSpawn.y, chosenSpawn.z, randomRotation)
     else
-        printf('No spawn points defined for mob %s (%u) in spawnPoints.', mob:getName(), mob:getID())
+        -- TODO This needs to stay here until all NMs have been updated to use entity.spawnPoints and the nm_spawn_points sql table is gone
+        if not UpdateNMSpawnPoint(mob:getID()) then
+            printf('No spawn points defined for mob %s (%u) in spawnPoints.', mob:getName(), mob:getID())
+        end
     end
 end
 
 -- potential lottery placeholder was killed
-xi.mob.phOnDespawn = function(ph, phList, chance, cooldown, params)
+xi.mob.phOnDespawn = function(ph, phNmId, chance, cooldown, params)
```

Comment:
And for this one I think it would be...
```lua
---@param ph CBaseEntity
---@param phNmId integer
---@param chance integer
---@param cooldown integer
---@param params table?
xi.mob.phOnDespawn = function(ph, phNmId, chance, cooldown, params)
```